source/Octopus.Client/Repositories/Async/BasicRepository.cs

```diff
@@ -37,25 +37,35 @@ protected BasicRepository(IOctopusAsyncRepository repository, string collectionL
         public IOctopusAsyncClient Client { get; }
         public IOctopusAsyncRepository Repository { get; }
 
-        protected virtual void CheckSpaceResource(IHaveSpaceResource spaceResource)
+        protected virtual async Task CheckSpaceResource(IHaveSpaceResource spaceResource)
         {
-            Repository.Scope.Apply(
+            await Repository.Scope.Apply(
                 whenSpaceScoped: space =>
                 {
                     if (spaceResource.SpaceId != null && spaceResource.SpaceId != space.Id)
                         throw new ResourceSpaceDoesNotMatchRepositorySpaceException(spaceResource, space);
+                    return Task.FromResult("");
```

Comment:
[nit] I believe it is more common to do `Task.FromResult(null)`

Reply:
`Task.FromResult(null)` would only work as `Task.FromResult<object>(null)`. I have used another common approach `Task.FromResult(0)`